feature: Add private zone support

mprimeaux opened the pull request:
## what

* Corrects `soa_enabled` logic to prevent creation of SOA records for private Route 53 zones.
* Ensures SOA records are only created for public zones (no associated `vpc_id`).

## why

* Private zones automatically include an SOA record managed by AWS; creating one manually causes errors.
* Public zones still require an explicit SOA record for consistent zone initialization.

Here's the error:

```shell
Error: creating Route53 Record: operation error Route 53: ChangeResourceRecordSets, 
https response error StatusCode: 400, RequestID: 9f01b1c0-9902-43bf-9d12-d073087872cc,
InvalidChangeBatch: [Invalid Resource Record: 'FATAL problem: DomainLabelEmpty (Domain label is empty) encountered with 'ns-0.awsdns-00.com.'', 
A HostedZone must contain exactly one SOA record.]
```

## references

* AWS Docs: [[SOA and NS records in Route 53 private hosted zones](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/SOA-NSrecords.html)](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/SOA-NSrecords.html)
* Closes #36 and #53

bartelemi reviewed (approved):
Nice one, would love to see this merged.

I've been waiting for this for years, but it was never a top priority for us 😄

mprimeaux committed:
feat(route53): add support for VPC associations in private zones

- Introduced new vpc_ids input to allow associating private hosted zones with one or more VPCs
- Updated dynamic "vpc" block to iterate over var.vpc_ids (empty list results in a public zone)
- Adjusted soa_enabled logic to treat empty vpc_ids as a public zone indicator
- Maintains plan-safe behavior and clear public/private zone separation

mprimeaux commented:
@bartelemi Is there anything I can do to assist in getting through this PR?

mprimeaux commented:
For those interested, my fork is in production and has been thoroughly tested with the related latest AWS service.
